### PR TITLE
AUT-137 CRD updates for approval of actions

### PR DESCRIPTION
**Pull request by harsh-px**
- AUT-138 Add needsApproval enforcement type in AutopilotRule
- AUT-137 Add new actionApprovals section in the AutopilotRuleObject to capture action approvals
- removed unused constants

Signed-off-by: Harsh Desai <harsh@portworx.com>

### CODE REVIEW

File: pkg/apis/autopilot/v1alpha1/autopilotrule.go [hunk 1]
```diff
@@ -4,8 +4,12 @@ import (
 	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// LabelSelectorOperator is the set of operators that can be used in a selector requirement.
-type LabelSelectorOperator string
+type (
+	// LabelSelectorOperator is the set of operators that can be used in a selector requirement.
+	LabelSelectorOperator string
+	// ActionApprovalState is the enum for approval states can object can take for it's actions
```
Comment (adityadani) [inline pkg/apis/autopilot/v1alpha1/autopilotrule.go]: nit: that an object can take

File: pkg/apis/autopilot/v1alpha1/constants.go [hunk 2]
```diff
@@ -1,78 +1,9 @@
...
-
-	// PolicyActionVolumeResize is an action to resize volumes
-	PolicyActionVolumeResize = "resize"
-
-	/***** Node actions *****/
-
-	// PolicyActionNodeRebalance is an action to rebalance a node
-	PolicyActionNodeRebalance = PolicyActionNode + actionRebalance
+	//  ApprovalRequired specifies that all actions for the rule will require an approval
+	ApprovalRequired EnforcementType = "approvalRequired"
```
Comment (adityadani) [inline pkg/apis/autopilot/v1alpha1/constants.go:8]: nit: the extra space is causing travis lint to fail.

**Review by adityadani — APPROVED** [review]
lgtm

**Comment by harsh-px** [comment]
Latest travis build has passed: https://travis-ci.org/github/libopenstorage/autopilot-api/builds/672247268

Travis or github bug causing status to not get updated here

**Pull request merged**